src/service/url-replacements-impl.js

```diff
@@ -812,6 +813,30 @@ export class UrlReplacements {
   }
 
   /**
+   * Append URL parameters from data-amp-addparams to URL
+   * @param {string} url
+   * @param {!Element} element An anchor element.
+   * @return {string} url
+   * @private
+   */
+  extendURLParams_(url, element) {
+    const additionalURLParamaters = element.getAttribute('data-amp-addparams');
+    if (!additionalURLParamaters) {
+      return url;
+    }
+    additionalURLParamaters.trim().split(/\s+/).forEach(urlParameter => {
```

Comment:
We'll need this to use [`addParamsToUrl`](https://github.com/rajkumarsrk/amphtml/blob/1db53de4d7b66cd225aa71da98d7482ff0c489b5/src/url.js#L171-L180), but unfortunately we'll need to do some parsing.

```js
// This should probably live in src/url.js
function spaceDelimitedParamsToMap(params) {
  return parseQueryString(params.replace(/\s+/, '&'));
}

// Here, in this code
return addParamsToUrl(url, spaceDelimitedParamsToMap(additionalURLParamaters));